src/app/clusters/basic-information/BasicInformationCluster.cpp

```diff
@@ -45,10 +41,36 @@ using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::BasicInformation;
 using namespace chip::app::Clusters::BasicInformation::Attributes;
 using namespace chip::DeviceLayer;
-using chip::Protocols::InteractionModel::Status;
 
 namespace {
 
+// fixed size for location strings
+static constexpr size_t kFixedLocationLength = 2;
+
+constexpr DataModel::AttributeEntry kMandatoryAttributes[] = {
```

Comment:
Followup: we should really figure out what the actual needs are around this "disable the mandatory attribute" business and why.  Duplicating all of this is very fragile; it would be nice if we could just use the code-generated mandatory attribute list....

If we really do need this, I would prefer that we added some sort of special-case to codegen to be able to flag mandatory attributes to be left out of its generated list (either hardcoded for this one, or based on some config file or something), so we don't have to duplicate things here.

Reply:
I will add some docs. Essentially specification decided to flip UniqueID from optional to mandatory and for backwards compatibility we currently support not having a uniqueID ... we respect this in our codegen integration and here.